Spokestack/NLUTensorflowSlotParsers.swift

```diff
@@ -32,15 +46,32 @@ internal struct NLUTensorflowSlotParser {
                     result.merge(slotToken, uniquingKeysWith: { $0 + $1 })
                 }
             })
-            // send each tag's tokens through the slot parser and return the result
-            .reduce(into: [:] as [String:Slot], { result, dictionaryEntry in
-                let (tag, whitespaceIndices) = dictionaryEntry
-                let slot = intent.slots.filter({ $0.name == tag }).first
-                if let s = slot, let slotValue = try self.slotFacetParser(slot: s, whitespaceIndices: whitespaceIndices, encoder: encoder, encodedTokens: encodedTokens) {
-                    result[tag] = Slot(type: s.type, value: slotValue)
-                }
-            })
-        return slots.isEmpty ? nil : slots
+    }
+    
+    /// Return a fixed set of slots defined by the intent, with values determined by the zip of model output tags to input tokens.
+    /// - Parameters:
+    ///   - intent:  The predicted intent.
+    ///   - tagsToTokens: Zip of tags (ignoring the "o" tag) and the index of the whitespaced encoded tokens.
+    ///   - encoder: The tokenizer instance to decode the `encodedTokens`.
+    ///   - encodedTokens: The tokens and associated metadata to decode for slot values.
+    /// - Throws: NLUError if the model output indicates a slot value that is not parseable.
+    /// - Returns: A structured `[intent : Slot]` dictionary of intent name keys with the slots defined by the intent and the slot's values as determined by the model output.
+    private func intentSlotMap(intent: NLUTensorflowIntent, tagsToTokens: [String : [Int]], encoder: BertTokenizer, encodedTokens: EncodedTokens) throws -> [String:Slot]? {
+        // Go over the slots defined by the intent, checking if each one has a value in the model output.
+        try intent.slots.reduce(into: [:], { (result, intentSlot) in
+            // Check the dictionary of slot name keys with input token values to see if there's a match for this intent slot.
+            let taggedSlotValues = tagsToTokens.filter({ $0.key == intentSlot.name })
+            if taggedSlotValues.isEmpty {
+                // no match, so hyrdate this intent slot with nil
```

Comment:
```suggestion
                // no match, so hydrate this intent slot with nil
```
The judges would also have accepted "include an empty slot"

Reply:
I like the judge here